### PR TITLE
fix: gpu support for rp Dockerfile

### PR DESCRIPTION
**Pull request by walkah**
### Review Type Requested (choose one):
- [X] **Logic** - thorough check (from everybody doing review)

### Summary
Our published docker image didn't support nvidia gpus for the PoW calculations. This update modifies the Dockerfile to have a `COMPUTE_MODE` build arg that allows `gpu` or `cpu` (with `gpu` as the default). For non-gpu machines (local dev, integration tests) it uses `cpu` in the `docker-compose.dev.yml` file.

### Task/Issue reference

Closes:  https://github.com/Lilypad-Tech/internal/issues/199

### CODE REVIEW

**Review by logan-wrld — APPROVED** [review]
Happy to see this! Now I can get your help on running local lilypad jobs! @walkah

**Review by logan-wrld — APPROVED** [review]
lgtm

File: docker/resource-provider/Dockerfile [hunk 1]
```diff
@@ -28,7 +24,17 @@ COPY --from=golang:1.22.4-alpine /usr/local/go/ /usr/local/go/
 ENV PATH="/usr/local/go/bin:${PATH}"
 
 COPY . .
-RUN go build -v .
+
+FROM base AS build-gpu
+RUN nvcc --version &&  nvcc --ptx -o ./pkg/resourceprovider/cudaminer/keccak.ptx  ./pkg/resourceprovider/cudaminer/keccak.cu
+RUN go build -v --tags cuda
+ENV DISABLE_POW=false
```
Comment (alvin-reyes) [inline docker/resource-provider/Dockerfile:31]: Why did we need to set the DISABLE_POW here?

Comment (walkah) [reply docker/resource-provider/Dockerfile:31]: This gets passed to `lilypad resource-provider` as `--disable-pow` (we disable pow for CPU ... i.e. local dev / integration tests)

**Pull request merged**